src/suggestion.rs

```diff
@@ -135,6 +158,131 @@ impl<'s> fmt::Display for Suggestion<'s> {
             0
         };
 
+        // For long lines, we will trim the literal displayed to fit in the terminal
+        // The misspelled word shall always be shown with as much info as possible
+        // Misspelled words that are too long shall also be ellipsized
+
+        // The paddings give some space for the ` {} ...` and extra indentation and formatting:
+        // ```
+        // 65 |  ...  here, on the second most pointless site ever! Well, wander ont  ...
+        //    |                                                                  ^^^
+        //    |                                                                                                                                                                                                ^^^
+        //    | - not, Ont, int, on, onto, cont, font, or one of 9 others
+        //    |
+        //    |   Possible spelling mistake found.
+        //    |
+        // ```
+        const PADDING_OFFSET: usize = 6;
+        const PADDING_OFFSET_START: usize = 4;
+        const PADDING_END: usize = 15;
+        const TOO_LONG_WORD: usize = 20;
+        const DISPLAYED_LONG_WORD: usize = 4usize;
+
+        let terminal_size: usize = get_terminal_size();
+        // We will be using ranges to help doing the fitting:
+        //
+        // |----------------------------------literal_word----------------------------------|
+        // |----------------------|---------misspelled_word---------|-----------------------|
+        // |-----left_context-----|---start_word----|----end_word---|-----right_context-----|
+        //
+        let mut range_left_context = Range {
+            start: 0usize,
+            end: marker_range_relative.start,
+        };
+        let mut range_right_context = Range {
+            start: marker_range_relative.end,
+            end: self.literal.as_str().chars().count(),
+        };
+        let mut range_start_word = Range {
+            start: marker_range_relative.start,
+            end: marker_range_relative.start,
+        };
+        let mut range_end_word = Range {
+            start: marker_range_relative.end,
+            end: marker_range_relative.end,
+        };
+        if self.literal.as_str().chars().count() > terminal_size {
+            let mut misspelled_word = format!(
+                "{}",
+                self.literal.sub(Range {
+                    start: marker_range_relative.start - 1,
+                    end: marker_range_relative.end
+                })
+            );
+            if marker_size > TOO_LONG_WORD {
+                range_start_word.start = marker_range_relative.start - 1;
+                range_start_word.end = range_start_word.start + DISPLAYED_LONG_WORD;
```

Comment:
```suggestion
                range_start_word = Range {
                     start = marker_range_relative.start - 1,
                     end = range_start_word.start + DISPLAYED_LONG_WORD
                };
```
easier on the eyes :)